src/Nethermind/Nethermind.Blockchain/ChainHeadInfoProvider.cs

```diff
@@ -27,14 +29,19 @@ public ChainHeadInfoProvider(ISpecProvider specProvider, IBlockTree blockTree, I
         {
         }
 
-        public ChainHeadInfoProvider(IChainHeadSpecProvider specProvider, IBlockTree blockTree, IReadOnlyStateProvider stateProvider, ICodeInfoRepository codeInfoRepository)
+        public ChainHeadInfoProvider(IChainHeadSpecProvider specProvider, IBlockTree blockTree, IReadOnlyStateProvider stateProvider, ICodeInfoRepository codeInfoRepository, bool forceSynced = false)
```

Comment:
as `forceSynced` is only used in test maybe either make it internal init property or add this logic in derived test-only type?

Reply:
Done